src/CTA.Rules.Actions/ActionHelpers/ConfigMigrate.cs

```diff
@@ -67,76 +71,105 @@ private XDocument LoadWebConfig(string projectDir)
-                    var value = connectionString.Attributes()?.First(a => a.Name?.ToString().ToLower() == Constants.connectionstring).Value?.Replace(@"\", @"\\");
+                var defaultContent = JsonConvert.DeserializeObject<JObject>(templateContent);
 
-                    config.Append("\"").Append(name).Append("\"").Append(":").Append("\"").Append(value).Append("\"").Append(",");
+                if (connectionStringsObjects.Count > 0)
+                {
+                    AddToJsonObject(defaultContent, Constants.ConnectionStrings, connectionStringsObjects);
                 }
-                catch (Exception ex)
+                if (appSettingsObjects.Count > 0)
                 {
-                    LogHelper.LogError(ex, "Error while parsing connection string");
+                    AddToJsonObject(defaultContent, Constants.appSettings, appSettingsObjects);
                 }
-            });
-
-            //This means, we found at least one connection string:
-            if (config.Length > 0)
+                return defaultContent;
+            }
+            else
             {
-                config.Remove(config.Length - 1, 1);
-                config.Insert(0, "\": { ");
-                config.Insert(0, Constants.ConnectionStrings);
-                config.Insert(0, "\"");
-                config.Append("},");
+                return null;
             }
+        }
 
-            webConfig.Descendants()?.First(d => d.Name?.ToString().ToLower() == Constants.appSettings)?.Descendants()?.ToList().ForEach(appSetting =>
+        /// <summary>
+        /// Gets connection strings from web.config file
+        /// </summary>
+        /// <param name="webConfig"></param>
+        /// <returns></returns>
+        private Dictionary<string, string> GetConnectionStrings(Configuration webConfig)
+        {
+            Dictionary<string, string> connectionStringObjects = new Dictionary<string, string>();
+            var connectionStrings = webConfig.ConnectionStrings?.ConnectionStrings;
+
+            if (connectionStrings != null)
             {
-                try
+                foreach (ConnectionStringSettings connectionString in connectionStrings)
                 {
-                    var key = appSetting.Attributes()?.First(a => a.Name?.ToString().ToLower() == Constants.key).Value;
-                    var value = appSetting.Attributes()?.First(a => a.Name?.ToString().ToLower() == Constants.value).Value;
-                    if (!Constants.appSettingsExclusions.Contains(key))
-                    {
-                        config.Append("\"").Append(key).Append("\"").Append(":").Append("\"").Append(value).Append("\"").Append(",");
-                    }
+                    connectionStringObjects.Add(connectionString.Name, connectionString.ConnectionString);
+                    _hasData = true;
```

Comment:
If taking other suggestion about assigning `_hasData` in `ProcessWebConfig()` method, we can remove this. If not, ignore this comment.